src/Microsoft.Extensions.Configuration.DockerSecrets/DockerSecretsConfigurationBuilderExtensions.cs

```diff
@@ -0,0 +1,41 @@
+using System;
+using Microsoft.Extensions.Configuration.DockerSecrets;
+
+namespace Microsoft.Extensions.Configuration
+{
+    /// <summary>
+    /// Extension methods for registering <see cref="DockerSecretsConfigurationProvider"/> with <see cref="IConfigurationBuilder"/>.
+    /// </summary>
+    public static class DockerSecretsConfigurationBuilderExtensions
```

Comment:
Discussed with @glennc today, he doesn't feel strongly that we need optional but since I added it, we can keep it